sql/update.go

```diff
@@ -333,6 +333,9 @@ func (p *planner) Update(n *parser.Update, autoCommit bool) (planNode, *roachpb.
 				if _, ok := colIDSet[col.ID]; ok {
 					continue
 				}
+				if _, ok := primaryKeyCols[col.ID]; ok {
+					continue
+				}
```

Comment:
completely unrelated, but does the comment in 331 mean that when updating a row, we always have to pull all of the old data out (i.e. does `UPDATE tbl SET onefield = 'foo' WHERE pk = 3`)? I would expect that to translate into a single write-only batch (similar to I think, an INSERT) which does `CPut(primaryKey, val=sentinel, expval=sentinel), Put(colKey, value)`.